hack/images/kubeone-e2e/release.sh

```diff
@@ -16,7 +16,7 @@
 
 set -euox pipefail
 
-TAG=v0.1.30
+TAG=v0.1.31
```

Comment:
v0.1.31 tag is already used on the main branch. Let's add a suffix to avoid conflicts, such as v0.1.31-go-1.19.